pkg/kubelet/util/pluginwatcher/example_handler.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pluginwatcher
+
+import (
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"golang.org/x/net/context"
+
+	v1beta1 "k8s.io/kubernetes/pkg/kubelet/util/pluginwatcher/example_plugin_apis/v1beta1"
+	v1beta2 "k8s.io/kubernetes/pkg/kubelet/util/pluginwatcher/example_plugin_apis/v1beta2"
+)
+
+type exampleHandler struct {
+	registeredPlugins        map[string]bool
+	mutex                    sync.Mutex
+	chanForHandlerAckTesting chan error // for testing
+}
+
+func NewExampleHandler() *exampleHandler {
+	return &exampleHandler{
+		chanForHandlerAckTesting: make(chan error),
+		registeredPlugins:        make(map[string]bool),
+	}
+}
+
+func (h *exampleHandler) Cleanup() error {
+	h.mutex.Lock()
+	defer h.mutex.Unlock()
+	h.registeredPlugins = make(map[string]bool)
+	return nil
+}
+
+func (h *exampleHandler) Handler(pluginName string, endpoint string, versions []string, sockPath string) (error, chan bool) {
+
+	// check for supported versions
+	if equal := reflect.DeepEqual([]string{"v1beta1", "v1beta2"}, versions); !equal {
```

Comment:
Hi @figo,

Suggest:

```go
if !reflect.DeepEqual([]string{"v1beta1", "v1beta2"}, versions); {
	return errors.New("not a supported version"), nil
}
```

The `equal` variable is not used anywhere inside the scope of the `if` block, so why declare the variable at all? Also, since you're not actually formatting a string, why use `fmt.Errorf`? Either change it to `errors.New` or consider `fmt.Errorf("invalid versions: %v", versions)`.